File: backend/supplier-service/src/main/java/com/ecotel/supplier_service/model/Supplier.java
package com.ecotel.supplier_service.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "supplier")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Supplier {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true, length = 20)
    private String supplierCode;

    @Column(nullable = false, length = 100)
    private String supplierName;

    @Column(length = 100)
    private String address;

    @Column(name = "contact_person", length = 50)
    private String contactPerson;

    @Column(length = 20)
    private String phone;

    @Column(length = 50)
    private String email;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL)
    private List<SupplierProduct> supplierProducts;
}

File: opt-frontend/src/types/inventory-opt/supplier.ts
export interface Supplier {
  id: string;
  supplierCode: string;
  supplierName: string;
  address: string;
  contactPerson: string;
  phone: string;
  email: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface SupplierRequest {
  supplierCode: string;
  supplierName: string;
  address: string;
  contactPerson: string;
  phone: string;
  email: string;
}

export interface SupplierUpdateRequest extends SupplierRequest {
  id: string;
}

export interface SupplierProduct {
  id: string;
  supplierId: string;
  supplierCode: string;
  supplierName: string;
  productId: string;
  productName: string;
  maxSupplyPerMonth: number;
  fixedOrderCost: number;
  unitPrice: number;
  committedLeadTimeDays: number;
  effectiveDate: string;
  isActive: boolean;
  notes: string;
  createdAt: string;
  updatedAt: string;
}

export interface SupplierProductRequest {
  productId: string;
  maxSupplyPerMonth: number;
  fixedOrderCost: number;
  unitPrice: number;
  committedLeadTimeDays: number;
  effectiveDate: string;
  notes?: string;
}

export interface SupplierProductUpdateRequest extends SupplierProductRequest {
  id: string;
}
